src/WebJobs.Extensions.OpenAI.CosmosDBSearch/CosmosDBSearchProvider.cs

```diff
@@ -16,10 +17,11 @@ sealed class CosmosDBSearchProvider : ISearchProvider
 {
     readonly IConfiguration configuration;
     readonly ILogger logger;
-    readonly int vectorSearchDimensions = 1536;
-    readonly int numLists = 1;
-    const string defaultSearchIndexName = "openai-index";
-    const string databaseName = "openai-database";
+    readonly IOptions<CosmosDBSearchConfigOptions> cosmosDBSearchConfigOptions;
+    readonly ConcurrentDictionary<string, MongoClient> cosmosDBClients = new();
+    string databaseName = "openai-functions-database";
+    string collectionName = "openai-functions-collection";
+    string indexName = "openai-functions-index";
```

Comment:
why do we need default values when this is mandatory for users to supply?

Reply:
Removed them.